src/dht/dht_server.cc

```diff
@@ -743,6 +743,11 @@ DhtServer::event_read() {
       if (!message[key_t].is_raw_string())
         throw dht_error(dht_error_protocol, "No transaction ID");
 
+	  //Check transaction id length to avoid use of it in dht_error if invalid
+	  if(message[key_t].as_string().size() > 20) {
+		  throw network_error();
+	  }
```

Comment:
In that case also add the length check to 'create_error' so that it doesn't try to copy the key.